tests/symbols-used.rs

```diff
@@ -0,0 +1,99 @@
+#![cfg(feature = "internal")]
+
+// This test checks that all symbols defined in Clippy's `sym.rs` file
+// are used in Clippy. Otherwise, it will fail with a list of symbols
+// which are unused.
+//
+// This test only runs when the `internal` feature is set, to allow
+// developers to temporarily define new, yet unused symbols without
+// failing the tests. Also, the test is a no-op if run as part of the
+// compiler test suite.
```

Comment:
IMO ditch the `cfg`, it's surprising that a test that doesn't require internals gets skipped without it

Reply:
Done.